corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -329,15 +316,30 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
                             value: value,
                         });
                         $field.trigger('change.select2');
-
-                        // recheck the check box if needed
-                        $($(".query-input-group")[i]).find('.search-for-blank').prop('checked', includeBlank);
                     }
                 }
                 self.setStickyQueryInputs();
             });
         },
 
+        toggleBlankSearch: function (e) {
+            var self = this,
+                searchForBlank = $(e.target).prop('checked'),
```

Comment:
Nit: I'm a proponent of always using `currentTarget` - doesn't matter here, but `target` can be a descendant of the element that has the event handler.

Reply:
thanks, I wasn't aware